### ISSUE TITLE
Cannot find primitive implementation for class A

### ISSUE THREAD

**Issue by loostro** [outworkers/phantom#751]
As a follow up on our chat with @alexflav23, it seems I've hit a corner case where implicit `Primitive[List[A]]` is not generated even though it should given an implicit `Primitive[A]` is in scope.

```
import com.outworkers.phantom.dsl._
import spray.json._
import pl.iterators.kebs.json.{KebsSpray, KebsEnumFormats}
 
case class CardNumber(value: String) extends AnyVal
case class CardStatus(value: String) extends AnyVal
case class CustomerId(value: UUID)   extends AnyVal
 
case class CustomerCard(number: CardNumber, status: CardStatus)
 
// removed other fields from "Customer" for brevity
case class Customer(id: CustomerId, cards: List[CustomerCard])
 
abstract class Customers
  extends Table[Customers, Customer]
  with KebsSpray.Snakified
  with KebsEnumFormats {
 
  // Kebs macros supply the implicit JsonReader[CustomerCard] and JsonWriter[CustomerCard]
  // used by the toJson and parseJson methods from spray.json package
  implicit val customerCardPrimitive: Primitive[CustomerCard] =
    Primitive.json[CustomerCard](_.toJson.toString)(_.parseJson.convertTo[CustomerCard])
 
  override def tableName: String = "customers"
 
  object id          extends UUIDColumn() with PartitionKey
  object cards       extends JsonListColumn[CustomerCard]()
}
```

Tried with versions: `2.13.5` and `2.14.4`, both produced the same results. The code above is a trimmed version. When trying to compile the actual project I am getting error:

```
[error] /{.. trimmed ...}/Customers.scala:28: Cannot find primitive implementation for class CustomerCard
[error]   object cards       extends JsonListColumn[CustomerCard]()
```

**Comment by loostro** [comment]
Also tried removeing the `extends AnyVal`:

```
case class CardNumber(value: String) //extends AnyVal
case class CardStatus(value: String) //extends AnyVal
```

Also tried simple types:

```
case class CustomerCard(number: String, status: String)
```

They produced the same compilation error.

**Comment by loostro** [comment]
As a workaround we've switched to `ListColum[String]` and we're mapping in `fromRow` and all `store/update` methods.

**Comment by alexflav23** [comment]
@loostro The macro call site is compiled outside of the table. In plain English you need to go one scope up, meaning the table should already have those implicits defined. Try a `package object` in your `dao` package or whatever, or a companion object definition.

**Comment by alexflav23** [comment]
@loostro Sorry for the delay in getting back to you, I just mean this:

```scala
object Formats {
   implicit val primitiveBla = Primitive.json...
}

import Formats._

abstract class MyTable extends Table ...
```

**Comment by loostro** [comment]
@alexflav23 No problem, appreciate your work. I will try if this fixes the problem as soon as I'm done with my current priority task and get back to you. Thanks.

**Comment by loostro** [comment]
@alexflav23 Indeed it compiles. Thank you!
